convert/gitlab/convert.go

```diff
@@ -256,6 +326,278 @@ func convertJobToStep(ctx *context, jobName string, job *gitlab.Job) []*harness.
 	return steps
 }
 
+// convertInheritDefaultFields converts the default fields from the default job into the current job.
+func convertInheritDefaultFields(spec *harness.StepExec, defaultJob *gitlab.Default, keys []string) {
+	for _, key := range keys {
+		switch key {
+		case "after_script":
+			if len(defaultJob.After) > 0 {
+				spec.Run = strings.Join(defaultJob.After, "\n")
+			}
+		case "before_script":
+			if len(defaultJob.Before) > 0 {
+				spec.Run = strings.Join(defaultJob.Before, "\n")
+			}
+		case "artifacts":
+			if defaultJob.Artifacts != nil {
+				//TODO no supported
+			}
+		case "cache":
+			if defaultJob.Cache != nil {
+				//TODO
+			}
+		case "image":
+			if defaultJob.Image != nil {
+				spec.Image, spec.Pull = convertImageAndPullPolicy(defaultJob.Image)
+			}
+		case "interruptible":
+			//TODO not supported
+		case "retry":
+			if defaultJob.Retry != nil {
+				//TODO not supported
+			}
+		case "services":
+			if len(defaultJob.Services) > 0 {
+				//TODO
+			}
+		case "tags":
+			if len(defaultJob.Tags) > 0 {
+				//spec.Tags = strings.Join(defaultJob.Tags, ", ") //TODO
+			}
+		case "duration":
+			//spec.Timeout = defaultJob.Timeout //TODO
+		default:
+			log.Printf("Unrecognized inherit key: %s", key)
```

Comment:
we should avoid using `log` since consumers of this library may want to bring their own logger

Reply:
if we need to handle logging, let's handle that in a separate PR where we come up with a logger strategy